LayoutKit/Views/ReloadableView.swift

```diff
@@ -80,12 +62,28 @@ extension UICollectionView: ReloadableView {
         registerClass(UICollectionReusableView.self, forSupplementaryViewOfKind: UICollectionElementKindSectionFooter, withReuseIdentifier: reuseIdentifier)
     }
 
-    public func insert(indexPaths indexPaths: [NSIndexPath]) {
-        insertItemsAtIndexPaths(indexPaths)
-    }
+    public func perform(batchUpdates batchUpdates: BatchUpdates) {
+        performBatchUpdates({
+            if batchUpdates.insertItems.count > 0 {
+                self.insertItemsAtIndexPaths(batchUpdates.insertItems)
+            }
+            if batchUpdates.deleteItems.count > 0 {
+                self.deleteItemsAtIndexPaths(batchUpdates.deleteItems)
+            }
+            for move in batchUpdates.moveItems {
+                self.moveItemAtIndexPath(move.from, toIndexPath: move.to)
+            }
 
-    public func insert(sections sections: NSIndexSet) {
-        insertSections(sections)
+            if batchUpdates.insertSections.count > 0 {
+                self.insertSections(batchUpdates.insertSections)
+            }
+            if batchUpdates.deleteSections.count > 0 {
+                self.deleteSections(batchUpdates.deleteSections)
+            }
+            for move in batchUpdates.moveSections {
+                self.moveSection(move.from, toSection: move.to)
+            }
+        }, completion: nil)
```

Comment:
The completion is only for when the animations finish. From a data integrity perspective, the updates are processed synchronously.

Unfortunately, there is no completion block for UITableView batch animations, so I chose to go with a behavior that would be consistent across UICollectionView and UITableView.